src/sources/journald.rs

```diff
@@ -181,6 +190,59 @@ impl JournaldConfig {
         }
         matches
     }
+
+    /// Builds the `schema::Definition` for this source using the provided `LogNamespace`.
+    fn schema_definition(&self, log_namespace: LogNamespace) -> Definition {
+        let schema_definition = match log_namespace {
+            LogNamespace::Vector => Definition::new_with_default_metadata(
+                Kind::bytes().or_null(),
+                [LogNamespace::Vector],
+            ),
+            LogNamespace::Legacy => Definition::new_with_default_metadata(
+                Kind::object(Collection::empty()),
+                [LogNamespace::Legacy],
+            ),
+        };
+
+        let mut schema_definition = schema_definition
+            .with_standard_vector_source_metadata()
+            // for metadata that is added to the events dynamically through the Record
+            .with_source_metadata(
+                JournaldConfig::NAME,
+                None,
+                &owned_value_path!("metadata"),
+                Kind::object(
+                    Collection::empty()
+                        .with_unknown(Kind::bytes())
+                        .with_known("message", Kind::bytes().or_null()),
```

Comment:
Would this still be inserted in the source metadata if it was present? I thought it would just be the value at root?

Reply:
It looks like if it's present you're removing it from the object there, so it wouldn't be a part of the schema there?

Reply:
:man_facepalming: